---
general:
  num_steps: 200
  save_rate: 200
  use_gpu: True
  print_debug: False
  use_vel: True
  save_figs: True
  render_figs: True
  flat_sensor: False
robot_control:
  # states: xyw # zw # yz #XYZ # XYZ #RPW # zw # this param was moved to test_env_vars.sh
  plot_states: xy
  horizon: 10
  num_target_samples: 2000 # note: more samples may be needed for higher dimensions
  num_traj_samples: 3000
  traj_buffer_capacity: 3000
  buffer_capacity: 3000
  use_magnitude: False
  R: 0.5 # [0.001,0.001,0.001,0.5,0.5,0.5] # 0.5 # [1.,1.,0.01]
robot_workspace: 
  sensor_z_offset: 0.15875 # cm (2.5" f/t + 3.75" camera = 6.25")
  sensor_mass: 5.0 
  max_force: 30.
  raw_states: xyzrpwb
  tray_ctrl_lim:
    - [-0.1, 0.1] # x
    - [-0.1, 0.1] # y
    - [-0.1, 0.1] # z
    - [-0.25, 0.25] # r
    - [-0.25, 0.25] # p
    - [-1.0, 1.0] # w
    - [-1.0, 1.0] # b 
  tray_lim:
    - [0.325, 0.625] # x
    - [-0.15, 0.15] # y
    # - [-0.3, 0.3] # 2y
    # - [0.25, 0.55] # z
    # - [0.35, 0.65] # z
    - [0.2, 0.5] # z (blocks) & offset ee (16.5cm)
    # - [0.15, 0.45] # z
    - [2.39, 3.89] # r
    # - [-0.75, 0.75] # r
    - [-0.75, 0.75] # p
    # - [-1., 1.] # w
    - [-2., 2.] # w
    - [0., 1.] # b
  robot_lim:
    - [-1.0, 1.0] # x
    - [-1.0, 1.0] # y
    # - [-2.0, 2.0] # 2y
    - [-1.0, 1.0] # z
    # - [2.39, 3.89] # r
    - [-0.75, 0.75] # r
    - [-0.75, 0.75] # p
    - [-1., 1.] # w
    # - [-2., 2.] # w
    # - [-0.5, 0.5] # w
    # - [-1., 1.] # w
    - [-1., 1.] # b
  robot_ctrl_lim:
    - [-1.25, 1.25] # x
    - [-1.25, 1.25] # y
    - [-1.25, 1.25] # z
    - [-0.5, 0.5] # r
    - [-0.5, 0.5] # p
    - [-1.25, 1.25] # w
    - [-1.5, 1.5] # b  
model:
  image_dim: [360,360,3] # rgb
  down_sample: 2
  zoom: 1
  z_dim: 16
  y_logvar_dim: 1 # [3,3]
  hidden_dim: [512,256]
  batch_size: 64
  CNNdict: # layers from outer to inner
    kernel_size: [3,3,5] 
    stride: [2,2,3] 
    channels: [10,10,20] 
  prior_steps: 0
  dx: False
trainer:
  model_lr: 0.001
  update_rate: 1
  frames_before_update: 1
  frames_before_training: 1 # 200
  num_learning_opt: 25 # reduce if doing synchronous updates (e.g. 3)
  target_learning_rate: 3
  gamma_weight: 0.1
  target_samples_scale: 1 # 1 means use the same as above, 2 means use 2x above, etc.
  hyperparam_ramp:
    fixed_beta: False
    fixed_gamma: False
    beta_manual_ramp: False
    beta_warmup_steps: 1000 # total number of beta updates
    beta_warmup_epoch: 10 # number of updates before beta can be updated
    beta_start_weight: 0.
    beta_end_weight: 0.05 # batch_size / num_steps
    gamma_manual_ramp: False
    gamma_warmup_steps: 1000 # total number of gamma updates
    gamma_warmup_epoch: 10 # number of updates before gamma can be updated
    gamma_start_weight: 0.
    gamma_end_weight: 1.0